voice_insights_summary.go

```diff
@@ -0,0 +1,154 @@
+	To              ToFrom            `json:"to"`
+	URL             string            `json:"url"`
+}
+
+type Attributes struct {
+	ConferenceParticipant bool `json:"conference_participant"`
+}
+
+type Edge struct {
+	Metrics struct {
+		Inbound  EdgeSummary `json:"inbound"`
+		Outbound EdgeSummary `json:"outbound"`
+	} `json:"metrics"`
+	Properties EdgeProperties `json:"properties"`
+}
+
+type EdgeSummary struct {
+	Codec                 int            `json:"codec"`
+	CodecName             string         `json:"codec_name"`
+	PacketsReceived       int            `json:"packets_received"`
+	PacketsSent           int            `json:"packets_sent"`
+	PacketsLost           int            `json:"packets_lost"`
+	PacketsLossPercentage float64        `json:"packets_loss_percentage"`
+	Jitter                MetricsSummary `json:"jitter"`
+	Latency               MetricsSummary `json:"latency"`
+	PacketDelayVariation  map[string]int `json:"packet_delay_variation"`
+}
+
+type EdgeProperties struct {
+	Direction           string `json:"direction"`
+	MediaRegion         string `json:"media_region"`
+	SignalingRegion     string `json:"signaling_region"`
+	TwilioMediaIP       string `json:"twilio_media_ip"`
+	TwilioSignalingIP   string `json:"twilio_signaling_ip"`
+	ExternalMediaIP     string `json:"external_media_ip"`
+	ExternalSignalingIP string `json:"external_signaling_ip"`
+	SIPCallID           string `json:"sip_call_id"`
+	UserAgent           string `json:"user_agent"`
+	SelectedRegion      string `json:"selected_region"`
+	Region              string `json:"region"`
+	DisconnectedBy      string `json:"disconnected_by"`
+	TrunkSID            string `json:"trunk_sid"`
+}
+
+type SdkEdge struct {
+	Metrics struct {
+		Inbound  SdkEdgeSummary `json:"inbound"`
+		Outbound SdkEdgeSummary `json:"outbound"`
+	} `json:"metrics"`
+	Properties SdkEdgeProperties `json:"properties"`
+	Events     SdkEdgeEvents     `json:"events"`
+}
+
+type SdkEdgeSummary struct {
+	EdgeSummary
+	Mos       MetricsSummary    `json:"mos"`
+	Rtt       MetricsSummary    `json:"rtt"`
+	Tags      map[string]string `json:"tags"`
+	AudioOut  MetricsSummary    `json:"audio_out"`
+	AudioIn   MetricsSummary    `json:"audio_in"`
+}
+
+type SdkEdgeProperties struct {
+	Direction string `json:"direction"`
+	Settings  struct {
+		DSCP              bool     `json:"dscp"`
+		ICERestartEnabled bool     `json:"ice_restart_enabled"`
+		Edge              string   `json:"edge"`
+		SelectedEdges     []string `json:"selected_edges"`
+	} `json:"settings"`
+}
+
+type SdkEdgeEvents struct {
+	Levels map[string]int `json:"levels"`
+	Groups map[string]int `json:"groups"`
+	Errors map[string]int `json:"errors"`
+}
+
+type MetricsSummary struct {
+	Max float64 `json:"max"`
+	Avg float64 `json:"avg"`
+	Min float64 `json:"min"`
+}
+
+type CallProperties struct {
+	Direction          Direction `json:"direction"`
+	DisconnectedBy     string    `json:"disconnected_by"`
+	PostDialDelay      int       `json:"pdd_ms"`
```

Comment:
Can we keep the unit? Or convert to a `time.Duration` (the latter is more work, though, I don't know that it is necessary.)

Reply:
I've changed this to `PostDialDelayMs`